src/test/fuzz.cpp

```diff
@@ -1,194 +1,186 @@
-    XDROutputFileStream out;
-    out.open(filename);
-    autocheck::generator<StellarMessage> gen;
-    for (size_t i = 0; i < n; ++i)
+    switch (fuzzerMode)
     {
+    case FuzzerMode::OVERLAY:
+    {
+        // HELLO, AUTH and ERROR_MSG messages cause the connection between
+        // the peers to drop. Since peer connections are only established
+        // preceding the persistent loop, a dropped peer is not only
+        // inconvenient, it also confuses the fuzzer. Consider a msg A sent
+        // before a peer is dropped and after a peer is dropped. The two,
+        // even though the same message, will take drastically different
+        // execution paths -- the fuzzer's main metric for determinism
+        // (stability) and binary coverage.
+        auto isDesiredXDR = [](StellarMessage& m) {
+            return m.type() != AUTH && m.type() != ERROR_MSG &&
+                   m.type() != HELLO;
+        };
+
+        XDROutputFileStream out;
+        out.open(filename);
+        autocheck::generator<StellarMessage> gen;
         try
         {
             StellarMessage m(gen(10));
+            while (!isDesiredXDR(m))
+                m = gen(10);
+
+            out.writeOne(m);
+        }
+        catch (xdr::xdr_bad_discriminant const&)
+        {
+        }
+        break;
+    }
+    case FuzzerMode::TRANSACTION:
+    {
+
+        // TODO: allow for more operations, not just account merge and
+        // native payments
+        auto predicate = [](Operation& op) {
+            return op.body.type() == ACCOUNT_MERGE ||
+                   (op.body.type() == PAYMENT &&
+                    op.body.paymentOp().asset.type() == ASSET_TYPE_NATIVE);
+        };
+
+        auto isDesiredXDR = [predicate](xdr::xvector<Operation>& ops) {
+            ops = std::accumulate(
+                ops.begin(), ops.end(), xdr::xvector<Operation>{},
+                [predicate](xdr::xvector<Operation> ops, Operation& op) {
+                    if (predicate(op))
+                        ops.emplace_back(op);
+
+                    return ops;
+                });
+
+            return ops.size() > 0;
+        };
+
+        XDROutputFileStream out;
+        out.open(filename);
+        autocheck::generator<xdr::xvector<Operation>> gen;
+        try
+        {
+            xdr::xvector<Operation> m(gen(10));
+            while (!isDesiredXDR(m))
+                m = gen(10);
+
             out.writeOne(m);
-            LOG(INFO) << "Message " << i << ": " << msgSummary(m);
         }
         catch (xdr::xdr_bad_discriminant const&)
         {
-            LOG(INFO) << "Message " << i << ": malformed, omitted";
         }
+        break;
+    }
     }
 }
+
+// default "tx" if unrecognized
+FuzzerMode
+getFuzzerModefromString(std::string const& fuzzerMode)
+{
+    if (iequals(fuzzerMode, "overlay"))
+    {
+        return FuzzerMode::OVERLAY;
+    }
+
+    if (iequals(fuzzerMode, "tx") || iequals(fuzzerMode, "transaction"))
```

Comment:
that seems not necessary because of default return value

Reply:
Good point, done.